syntaxes/src/expression/grammar.ts

```diff
@@ -0,0 +1,886 @@
+import { GrammarDefinition } from '../types';
+
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export const expression: GrammarDefinition = {
+  scopeName: 'expression.ng',
+  injectionSelector: 'L:text.html -comment',
+  patterns: [{ include: '#ngExpression' }],
+  repository: {
+    ngExpression: {
+      name: 'meta.expression.ng',
+      patterns: [
+        {
+          include: '#tsString',
+        },
+        {
+          include: '#literal',
+        },
+        {
+          include: '#ternaryExpression',
+        },
+        {
+          include: '#expressionOperators',
+        },
+        {
+          include: '#functionCall',
+        },
+        {
+          include: '#identifiers',
+        },
+        {
+          include: '#parenExpression',
+        },
+        {
+          include: '#punctuationComma',
+        },
+        {
+          include: '#punctuationAccessor',
+        },
+      ],
+    },
+
+    arrayLiteral: {
+      name: 'meta.array.literal.ts',
+      begin: /\[/,
+      beginCaptures: {
+        '0': {
+          name: 'meta.brace.square.ts',
+        },
```

Comment:
Sorry for the nitpick -- would you be willing to remove the quotes from the numeric keys and put the objects with only one key/value pair on a single line for consistency with the Template grammar? I think it makes it a little more readable, and consistency is nice -- but it's also kind of tedious, so I wouldn't blame you for leaving it as-is for now. :)

Reply:
I'll sure do it. I missed all of them. Got used to using prettier and completely forgot about it, it does all the dirty work. haha

Reply:
Removed from all of the numbers